app/metrics/admin.py

```diff
@@ -54,22 +57,70 @@ def save_model(self, request, obj, form, change):
 
 @admin.register(QuestionSet)
 class QuestionSetAdmin(ModelAdmin):
-    exclude = EDIT_TRACKING_FIELDS
     prepopulated_fields = {"slug": ["name"]}
 
     list_display = (
         "name",
         "user",
     )
 
+    def get_queryset(self, request):
+        qs = super().get_queryset(request)
+        if request.user.is_superuser:
+            return qs
+        
+        user_node = request.user.get_node()
+        if user_node:
+            # Filter QuestionSets based on the node
+            exclude = EDIT_TRACKING_FIELDS  # needs to be included if we want to be able to change user for a set
+            return qs.filter(
+                    node = user_node
+            ).distinct() | qs.filter(node = None).distinct()
+        return qs.none()
+
     def save_model(self, request, obj, form, change):
-        obj.user = request.user
+        if 'user' in form.cleaned_data:
+            obj.user = form.cleaned_data['user']
+        else:
+            obj.user = request.user
+        if not request.user.is_superuser:
+            obj.node = request.user.get_node()
         return super().save_model(request, obj, form, change)
+    
+    def has_change_permission(self, request, obj = None):
+        # Can only change sets from own node
+        if obj and not request.user.is_superuser:
+            user_node = request.user.get_node()  # Retrieve the user's associated node
+            if obj.node != user_node:
+                return False
+        return super().has_change_permission(request, obj=obj)
+
+    def has_delete_permission(self, request, obj = None):
+        # Node users cannot delete objects
+        if not request.user.is_superuser:
```

Comment:
Why can't we just use the normal permission structure here and not assign delete permissions?